### DISCUSSION TITLE
Add scroll area options to nicegui

### DISCUSSION THREAD

**Discussion by eli-kha** [zauberzeug/nicegui#571] [Ideas / Feature Requests]
Quasar offers [scroll-area](https://quasar.dev/vue-components/scroll-area) to creating scrollable sections.
I think that it would be good to add this as an element or embed it into the row/column elements with a flag `scrollable=True`.

What do you think?

**Comment by rodja** [comment]
Scroll areas would be really helpful. I think `ui.scrollable` or similar is a bit better than adding a parameter to row/column. That way the user as a dedicated element to set props as documented by Quasar.

Anyone interested in implementing this? Should be a "good first issue" to learn the code base.

**Reply by elik-vayyar** [reply]
I'm willing to work on it but it'll probably be a bit slow. I don't have much time to spare for this.

Before opening this discussion I played around with `ui.element('q-scroll-area' ) ` and for some reason it didn't render the content under it.
I probably need some guidance.

**Reply by falkoschindler** [reply]
@rodja I tend to stick with Quasar's terminology "scroll_area", which is a noun like other UI elements.

**Reply by falkoschindler** [reply]
@eli-kha You probably don't see the area at all because it has no default size. I tend to give it a fate background color to better see what's going on. Try something like this:

```py
with ui.element('q-scroll-area').classes('w-40 h-40 bg-gray-100'):
    for i in range(100):
        ui.label(f'Line {i}')
```

**Reply by eli-kha** [reply]
Just FYI @elik-vayyar is my other user. I accidently replied with it.

**Reply by eli-kha** [reply]
> @eli-kha You probably don't see the area at all because it has no default size. I tend to give it a fate background color to better see what's going on. Try something like this:
> 
> ```python
> with ui.element('q-scroll-area').classes('w-40 h-40 bg-gray-100'):
>     for i in range(100):
>         ui.label(f'Line {i}')
> ```

Thanks. It works for me as well.

This raises an implementation question regarding the future element behavior: 
- I'd expect the default to be for the scroll area to span the entire container it's in.
- We can do it with `w-full h-full`
- The problem is if the size is not set by the element above it doesn't work as expected.

example:
```python
with ui.card().classes('w-32 h-32'):
    with ui.element('q-scroll-area').classes('w-full h-full'):
        ui.label("""
        Lorem ipsum dolor sit amet, 
        consectetur adipiscing elit. 
        Suspendisse tincidunt nisl ac elit rutrum, 
        vitae gravida lorem tincidunt. 
        Vivamus auctor.
        """)

with ui.card():
    with ui.element('q-scroll-area').classes('w-full h-full'):
        ui.label("""
        Lorem ipsum dolor sit amet, 
        consectetur adipiscing elit. 
        Suspendisse tincidunt nisl ac elit rutrum, 
        vitae gravida lorem tincidunt. 
        Vivamus auctor.
        """)
```

I'll go over the Quasar docs and see how they do it there.

**Reply by falkoschindler** [reply]
Yes, that's an unfortunate. For `ui.aggrid` we simply use `width: 100%` and `height: 16rem` (see `.nicegui-aggrid` in nicegui.css) so that the user at least sees _something_.
Note that using plain CSS in `nicegui.css` instead of Tailwind classes avoids issues with conflicting classes when the user tries to override the height: An initial Tailwind class of, e.g., `h-40` might take precedence over a user-defined height of `h-20`. But `h-20` will correctly overrule any height defined in `nicegui.css`.

**Comment by eli-kha** [comment]
After a long time. I got around to this.
I might be aiming too high considering I don't really know JS but I wanted to expose the scroll event and later some of the methods of the component.

I worked mostly by referencing other elements: https://github.com/eli-kha/nicegui/tree/add-scroll-area-element-1
@falkoschindler or @rodja, I couldn't get the event to work and I'd appreciate some guidance.

**Reply by falkoschindler** [reply]
Thanks, @eli-kha! It looks like this is related to PR #1072. But why did you close it? If you're just unsure about the implementation, you might want to convert it to a draft PR. But closing it indicates it isn't relevant anymore.

**Comment by eli-kha** [comment]
merged in #1072

**Comment by nasischijf** [comment]
Does scroll_area have a maximum width of 80 and a height of 15 characters? When i make it 'w-81' it schrinks to 'w40'.  For my application I want a scrollable area of full width of the screen and as high as the half width of the screen.  Do I need to use another type of element than Scroll_area when I want to do this?

**Reply by falkoschindler** [reply]
"w-81" isn't a valid Tailwind class. See https://tailwindcss.com/docs/width for more information.
To use a custom value, you can write something like `.class('w-[300px]')` or `.style('width: 300px')`.

**Reply by nasischijf** [reply]
That's it! Thank you very much!
